### PR TITLE
Adding rules for templates

### PR DESCRIPTION
**Pull request by gagoar**
**Issue Reference**:  #66 

## Description

adding rules to validate the PR contains an Issue Reference, so it's always linked. 

## Motivation and Context

This is something that will save time when a PR is opened. 

## How Has This Been Tested?

Creating this PR and editing it. given that it reacts to opened and edited. 
some edit

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/template_validation_rules/validate_issue_is_present.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Rule to validate the the template has an issue attached",
```
Comment (cyamonide) [inline .github/template_validation_rules/validate_issue_is_present.json:2]:
```suggestion
  "name": "Rule to validate that the template has an issue attached",
```

File: .github/template_validation_rules/validate_issue_is_present.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "action": "comment",
+  "eventJsonPath": "$[?(@.body.match(/Issue Reference.*: #[0-9]{3}/) && @.user.login != 'renovate-bot')]",
+  "customMessage": "Thanks for contributing!. The pull request description looks good. Please give us some time to review!",
```
Comment (cyamonide) [inline .github/template_validation_rules/validate_issue_is_present.json:5]:
```suggestion
  "customMessage": "Thanks for contributing! The pull request description looks good. Please give us some time to review!",
```

File: .github/workflows/validate_template.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    name: Validate Template
+    runs-on: ubuntu-latest
+    steps:
+      - name: event location
+        run: |
+          echo $GITHUB_EVENT_PATH
+      - uses: actions/upload-artifact@master
+        with:
+          name: event
+          path: /home/runner/work/_temp/_github_workflow/event.json
```
Comment (cyamonide) [inline .github/workflows/validate_template.yml:18]: what does this do?
the absolute path into a temp directory makes me a little nervous

File: __mocks__/event.json [hunk 4]
```diff
@@ -33,7 +33,7 @@
       "type": "User",
       "site_admin": false
     },
-    "body": "This is a pretty simple change that we need to pull into master (ticket: 123).",
+    "body": "This is a pretty simple change that we need to pull into master\n This is a pretty simple change that we need to pull into master (**Issue Reference**: #123) ",
```
Comment (cyamonide) [inline __mocks__/event.json:36]: is this supposed to be duplicated?

File: __tests__/JSONPath.spec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { loadJSONFile } from '../src/util/loadJSONFile';
+import JSONPath from 'jsonpath';
+import { env } from '../src/environment';
+import { Event } from '../src/util/constants';
+describe('JsonPath', () => {
+  it('should match excluding user but matching title', () => {
+    // const pattern = "$..[?(@.body.match(/Issue Reference.*: #[0-9]{3}/) && @.user.login != 'renovate-bot')]";
```
Comment (cyamonide) [inline __tests__/JSONPath.spec.ts:7]:
```suggestion
```

File: .github/template_validation_rules/validate_issue_is_present.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Rule to validate the the template has an issue attached",
+  "action": "comment",
+  "eventJsonPath": "$[?(@.body.match(/Issue Reference.*: #[0-9]{3}/) && @.user.login != 'renovate-bot')]",
```
Comment (cyamonide) [inline .github/template_validation_rules/validate_issue_is_present.json:4]: Issue references may not always be 3 characters long - e.g. #58
Changing this to match one or more digits might be more suitable:
```suggestion
  "eventJsonPath": "$[?(@.body.match(/Issue Reference.*: #[0-9]+/) && @.user.login != 'renovate-bot')]",
```

File: __mocks__/required_rules/rule2.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "customMessage": "Thanks for contributing, everything looks good in your PR description, please give us some time to review.",
   "action": "comment",
-  "eventJsonPath": "$..[?(@.body.match('Issue: '))]",
+  "eventJsonPath": "$..[?(@.body.match(/Issue Reference.*: #[0-9]{4}/) && @.user.login != 'renovate-bot')]",
```
Comment (cyamonide) [inline __mocks__/required_rules/rule2.json:4]: Similar comment as earlier with the Issue reference quantifier being `{4}`, but this is a mock and should be fine

**Comment by gagoar** [comment]
> Looks good; left a few minor comments and questions here and there.
> 
> Question overall, do we want to be committing `dist/*` files, or should those be `.gitignore`d?

we wanna be committing it because the action uses the dist as a binary to run.

File: .github/template_validation_rules/validate_issue_is_present.json (hunk 6 above)
Comment (gagoar) [reply .github/template_validation_rules/validate_issue_is_present.json:4]: yea! that's correct, I was trying to make it fail. the final version should be + like you are offering.

File: .github/workflows/validate_template.yml (hunk 3 above)
Comment (gagoar) [reply .github/workflows/validate_template.yml]: yea, I tried to use the actual path from a variable but it didn't catch it, have you tried it ?

File: __mocks__/event.json (hunk 4 above)
Comment (gagoar) [reply __mocks__/event.json]: nope. good catch

File: .github/workflows/validate_template.yml (hunk 3 above)
Comment (cyamonide) [reply .github/workflows/validate_template.yml]: I have not tried it before 😬

Comment (cyamonide) [reply .github/workflows/validate_template.yml]: `github.event_path`!

**Pull request merged**